core/tests/protractor/editorAndPlayer.js

```diff
@@ -26,11 +26,69 @@ var ExplorationPlayerPage =
 
 describe('Full exploration editor', function() {
   var explorationPlayerPage = null;
+  var parentId2 = null;
+  var refresherExplorationId = null;
+  var currentExplorationId = null;
+  var pathname = null;
+  var parentId1 = null;
 
   beforeAll(function() {
     explorationPlayerPage = new ExplorationPlayerPage.ExplorationPlayerPage();
   });
 
+  it('should redirect back to parent exploration correctly when parent id is ' +
+      'given as query parameter', function() {
+    users.createUser('user1@editorAndPlayer.com', 'user1EditorAndPlayer');
+    users.login('user1@editorAndPlayer.com');
+
+    workflow.createAndPublishExploration(
+      'Parent Exploration 1',
+      'Algebra',
+      'This is the topmost parent exploration.');
+    general.getExplorationIdFromEditor().then(function(explorationId) {
+      parentId1 = explorationId;
+
+      workflow.createAndPublishExploration(
+        'Parent Exploration 1',
```

Comment:
Shouldn't this be "Parent Exploration 2"?

Reply:
Ya, changed it.